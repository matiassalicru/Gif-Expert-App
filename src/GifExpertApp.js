import React, { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

export const GifExpertApp = ({ defaultCategories= []}) => {
  const [categories, setCategories] = useState(defaultCategories);

  return (
    <>
      <h1 className="title animate__animated animate__lightSpeedInRight">
        Gifs to you
      </h1>
      <AddCategory setCategories={setCategories} />
      <hr />
      <ol>
        {categories.map((category) => (
          <GifGrid key={category} category={category} />
        ))}
      </ol>
    </>
  );
};
