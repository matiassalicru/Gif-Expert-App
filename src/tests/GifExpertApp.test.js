import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifExpertApp } from '../GifExpertApp';

describe('Pruebas con el gifExpertApp', () => {

    test('Debe de mathchear con el snapshot', () => {
        const wrapper = shallow(<GifExpertApp/>)

        expect(wrapper).toMatchSnapshot();
    })
    

    test('Debe de mostrar una lista de categorias', () => {
        
        const categories = ['The office', 'Friends'];

        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);

    })
    
})
