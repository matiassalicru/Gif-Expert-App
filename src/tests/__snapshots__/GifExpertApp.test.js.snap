// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con el gifExpertApp Debe de mathchear con el snapshot 1`] = `
<Fragment>
  <h1
    className="title animate__animated animate__lightSpeedInRight"
  >
    Gifs to you
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas con el gifExpertApp Debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h1
    className="title animate__animated animate__lightSpeedInRight"
  >
    Gifs to you
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="The office"
      key="The office"
    />
    <GifGrid
      category="Friends"
      key="Friends"
    />
  </ol>
</Fragment>
`;
